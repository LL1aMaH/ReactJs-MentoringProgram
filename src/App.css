@import './styles/colors.css';

:global(*) {
  box-sizing: border-box;
}

:global(*::before) {
  box-sizing: border-box;
}

:global(*::after) {
  box-sizing: border-box;
}

:global(html) {
  background-color: var(--primary-background-color);
}

body {
  margin: 0;
  padding: 0;
  width: calc(100vw - 14px);
}

::-webkit-scrollbar {
  width: 14px;
}

::-webkit-scrollbar-track {
  border-radius: 10px;
  box-shadow: inset 0 0 10px 10px transparent;
  border: solid 3px transparent;
}

::-webkit-scrollbar-thumb {
  border-radius: 10px;
  box-shadow: inset 0 0 10px 10px var(--secondary-dark-color);
  border: solid 3px transparent;
}
