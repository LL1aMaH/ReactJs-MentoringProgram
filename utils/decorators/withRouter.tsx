import React from 'react';

import { MemoryRouter } from 'react-router-dom';

export type TComponentFunc = () => JSX.Element;
export type TDecoratorLocation = {
  pathname?: string;
  search?: string;
};

const withRouter = (
  componentFunc: TComponentFunc,
  location: TDecoratorLocation = { pathname: '/' },
): JSX.Element => {
  return <MemoryRouter initialEntries={[location]}>{componentFunc()}</MemoryRouter>;
};

export default withRouter;
