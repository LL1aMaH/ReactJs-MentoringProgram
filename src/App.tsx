import React from 'react';

import './App.css';

const App = (): JSX.Element => <div>Part 2</div>;

export default App;
