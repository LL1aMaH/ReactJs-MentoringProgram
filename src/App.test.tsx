import React from 'react';
import { render } from '@testing-library/react';

import withRouter from '../utils/decorators/withRouter';

import App from './App';

describe('NotFoundPage', () => {
  it('renders correctly', () => {
    const component = render(withRouter(() => <App />));
    expect(component.asFragment()).toMatchSnapshot();
  });
});
