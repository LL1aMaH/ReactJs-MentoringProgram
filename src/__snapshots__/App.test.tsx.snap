// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage renders correctly 1`] = `
<DocumentFragment>
  <div
    class="preview"
    style="background-image: url(/src/Assets/pictures/cover-image.jpg); filter: blur(5px);"
  />
  <div
    class="previewContent"
  >
    <div
      class="buttons"
    >
      <a
        class="logo"
        href="/"
      >
        <b>
          netflix
        </b>
        roulette
      </a>
      <button
        class="button button"
        data-testid="addMovie"
        tabindex="0"
      >
        + ADD MOVIE
      </button>
    </div>
    <p>
      FIND YOUR MOVIE
    </p>
    <div
      class="searchBox searchBox"
    >
      <input
        placeholder="What do you want to watch?"
        value=""
      />
      <button
        class="button button"
        data-testid="button"
        tabindex="0"
      >
        SEARCH
      </button>
    </div>
  </div>
  <div
    class="navbar"
  >
    <div
      class="buttons"
    >
      <button
        class="button activeButton button"
        tabindex="0"
      >
        ALL
      </button>
      <button
        class="button button"
        tabindex="0"
      >
        DOCUMENTARY
      </button>
      <button
        class="button button"
        tabindex="0"
      >
        COMEDY
      </button>
      <button
        class="button button"
        tabindex="0"
      >
        HORROR
      </button>
      <button
        class="button button"
        tabindex="0"
      >
        CRIME
      </button>
    </div>
    <div
      class="dropDown"
    >
      <span
        class="label"
      >
        SORT BY
      </span>
      <div
        class="container dropdown"
      >
        <button
          class="selectButton select dropdownClose button"
          data-testid="selectButton"
          tabindex="0"
        >
          RELEASE DATE
          <svg />
        </button>
        <ul
          class="selectList close"
        >
          <li>
            <button
              class="option select selectOption"
              type="button"
            >
              RELEASE DATE
            </button>
          </li>
          <li>
            <button
              class="option select"
              type="button"
            >
              RATING
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    class="container"
  >
    <span>
      0 movies found
    </span>
    <div
      class="list"
    />
    ;
  </div>
</DocumentFragment>
`;
